src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/IDynamicInterfaceCastableSupport.cs

```diff
@@ -28,11 +35,32 @@ internal static IntPtr IDynamicCastableGetInterfaceImplementation(IDynamicInterf
             {
                 ThrowInvalidOperationException(implType);
             }
-            IntPtr result = RuntimeImports.RhResolveDispatchOnType(implType, interfaceType, slot);
+
+            MethodTable* genericContext = null;
+            IntPtr result = RuntimeImports.RhResolveDynamicInterfaceCastableDispatchOnType(implType, interfaceType, slot, &genericContext);
             if (result == IntPtr.Zero)
             {
                 IDynamicCastableGetInterfaceImplementationFailure(instance, interfaceType, implType);
             }
+
+            if (genericContext != null)
```

Comment:
This looks same as the delegate thunk pool. Should we just have one shared thunk pool?

Reply:
It feels more risky to touch the delegate thunk pool. We might need to service this to .NET 9. With that in mind would you still prefer we use the same pool?

Reply:
I do not have strong opinion. We can clean it up later. There is quite a bit leftovers from when the thunk pool was in the shared MRT.